service/s3/s3manager/download.go

```diff
@@ -402,18 +407,31 @@ func (d *downloader) downloadChunk(chunk dlchunk) error {
 	// Get the next byte range of data
 	in.Range = aws.String(chunk.ByteRange())
 
+	var (
+		w       io.Writer
+		cleanup func()
+	)
+
 	var n int64
 	var err error
 	for retry := 0; retry <= d.partBodyMaxRetries; retry++ {
+		if d.cfg.DownloadStrategy != nil {
```

Comment:
Does it make sense for this loop's body to be broken out into its own function so that the `cleanup` function can be called via a `defer`? Also might be easier to read if `w` and `cleanup` are defined within the loop body to prevent any confusion on if they are used outside the loop.